tar option failing to pull image layer

I have the container-structure-test working on xubuntu 16.04 64 bit for a couple of simple file existence tests for the default docker driver but I haven't been able to get the tar driver to pull the image.
I'm getting errors like:
ERRO[0110] Failed to pull image layer: Get https://dseasb33srnrn.cloudfront.net/registry-v2/docker/registry/v2/blobs...

I've tried on different machines and different locations.
Is there a trick to getting the tar option to work? 

My command line is:
sudo ./container-structure-test -test.v -driver tar -image python python_file_test.yaml

Hey @nkubala yes I'm expecting it to just do a docker pull. If I use the docker driver instead of the tar driver it works fine. I'm not using a private registry. 
On my home Xubuntu laptop the end of the output looks like this:
ERRO[0152] remove /tmp/python638756840/var/lib/dpkg/info/.wh.libtimedate-perl.md5sums: no such file or directory 
INFO[0162] Finished prepping image python               
--- FAIL: TestAll (162.11s)
    --- PASS: TestAll/File_Existence_Test:_Root (55.16s)
    --- FAIL: TestAll/File_Existence_Test:_python (53.54s)
    	structure.go:94: File /usr/local/bin/python should exist but does not, got error: stat /tmp/python905351533/usr/local/bin/python: no such file or directory!
    --- PASS: TestAll/Metadata_Test (53.40s)
	structure_test.go:49: Total tests run: 2
FAIL

If I was completing this test with dgoss I'd expect it to take less than a second.
Perhaps there could be an option to use a local container rather than trying to pull it for the tar driver?

--- a/Gopkg.lock
+++ b/Gopkg.lock
@@ -16,7 +16,7 @@
     "cmd/util/output",
     "pkg/util"
   ]
-  revision = "d2b317ec63b9e9db4ad6a4065399d74c9b7ebd78"
+  revision = "9ba9d88a7caac4d74d4ca39700b5c62331005785"
 
 [[projects]]
   name = "github.com/Microsoft/go-winio"
@@ -245,6 +245,6 @@
 [solve-meta]
   analyzer-name = "dep"
   analyzer-version = 1
-  inputs-digest = "b960156035e417f320da5604f8883a6e452d54bcacf73e536d8c9687ea977977"
+  inputs-digest = "204c01cd5f22633661ed23946429c841b3d3f2849540f24e8b2b4b1beb5503f1"
   solver-name = "gps-cdcl"
   solver-version = 1

--- a/Gopkg.toml
+++ b/Gopkg.toml
@@ -23,7 +23,7 @@
 
 [[constraint]]
   name = "github.com/GoogleCloudPlatform/container-diff"
-  revision = "d2b317ec63b9e9db4ad6a4065399d74c9b7ebd78"
+  revision = "9ba9d88a7caac4d74d4ca39700b5c62331005785"
 
 [[constraint]]
   name = "github.com/fsouza/go-dockerclient"

--- a/drivers/tar_driver.go
+++ b/drivers/tar_driver.go
@@ -31,21 +31,29 @@ type TarDriver struct {
 }
 
 func NewTarDriver(args DriverConfig) (Driver, error) {
-	// if the image is in the local daemon, we should be using the docker driver anyway.
-	// only try remote.
 	var prepper pkgutil.Prepper
+	var image pkgutil.Image
+	var err error
 	imageName := args.Image
 	if pkgutil.IsTar(imageName) {
-		prepper = pkgutil.TarPrepper{
+		prepper = &pkgutil.TarPrepper{
 			Source: imageName,
 		}
 	} else {
-		prepper = pkgutil.CloudPrepper{
+		// see if image exists locally first.
+		prepper = &pkgutil.DaemonPrepper{
 			Source: imageName,
+		}
+		image, err = prepper.GetImage()
+		if err != nil {
+			// didn't find locally, try to pull from remote.
+			prepper = &pkgutil.CloudPrepper{
+				Source: imageName,
+			}
+			image, err = prepper.GetImage()
 		}
 	}
 
-	image, err := prepper.GetImage()
 	if err != nil {
 		// didn't find image anywhere; exit
 		return nil, err

